internal/xds/rbac/rbac_engine.go

```diff
@@ -108,13 +108,13 @@ type engine struct {
 // newEngine creates an RBAC Engine based on the contents of policy. Returns a
 // non-nil error if the policy is invalid.
 func newEngine(config *v3rbacpb.RBAC) (*engine, error) {
-	a := *config.Action.Enum()
+	a := *config.GetAction().Enum()
```

Comment:
`a := config.GetAction()` ?

`Enum()` should only be used when setting a value in a message field.  This is allocating new memory, returning a pointer to it, then dereferencing it.

Reply:
Ah, I see. Changed.